File: backend/infraction-service/app/com/radovan/play/controllers/PrometheusController.scala
package com.radovan.play.controllers

import io.micrometer.prometheusmetrics.PrometheusMeterRegistry
import jakarta.inject.Inject
import play.api.mvc.{AbstractController, Action, AnyContent, ControllerComponents}

class PrometheusController @Inject()(
                                      cc: ControllerComponents,
                                      prometheusRegistry: PrometheusMeterRegistry
                                    ) extends AbstractController(cc) {

  def getMetrics: Action[AnyContent] = Action {
    val scrapeData = prometheusRegistry.scrape()
    Ok(scrapeData).as("text/plain")
  }
}

File: backend/registration-service/app/com/radovan/play/controllers/PrometheusController.java
package com.radovan.play.controllers;

import io.micrometer.prometheusmetrics.PrometheusMeterRegistry;
import jakarta.inject.Inject;
import play.mvc.Controller;
import play.mvc.Result;

public class PrometheusController extends Controller {

    private PrometheusMeterRegistry registry;

    @Inject
    private void initialize(PrometheusMeterRegistry registry){
        this.registry = registry;
    }

    public Result getMetrics(){
        return ok(registry.scrape());
    }
}
